lighthouse-core/report/html/renderer/util.js

```diff
@@ -336,6 +350,51 @@ class Util {
     };
   }
 
+  /**
+   * @param {string|OriginalURL} value
+   * @return {OriginalURL}
+   */
+  static createOrReturnURL(value) {
+    if (value instanceof URL) {
+      return value;
+    }
+
+    return new URL(/** @type {string} */ (value));
+  }
+
+  /**
+   * Gets the tld of a domain
+   *
+   * @param {string} hostname
+   * @return {string} tld
+   */
+  static getTld(hostname) {
+    const tlds = hostname.split('.').slice(-2);
+
+    if (!listOfTlds.includes(tlds[0])) {
+      return `.${tlds[tlds.length - 1]}`;
+    }
+
+    return `.${tlds.join('.')}`;
+  }
+
+  /**
+   * Returns a primary domain for provided hostname (e.g. www.example.com -> example.com).
+   * @param {string|OriginalURL} url hostname or URL object
+   * @returns {string}
+   */
+  static getRootDomain(url) {
+    const hostname = Util.createOrReturnURL(url).hostname;
+    const tld = Util.getTld(hostname);
+
+    // tld is .com or .co.uk which means we means that length is 1 to big
+    // .com => 2 & .co.uk => 3
+    const splitTld = tld.split('.');
+
+    // get TLD + root domain
+    return hostname.split('.').slice(-splitTld.length).join('.');
```

Comment:
I think this new logic breaks on `localhost`, could we add a quick test and fix?

Reply:
it's actually working :)

```
// hostname = "localhost"
const hostname = Util.createOrReturnURL(url).hostname;
// ".localhost"
const tld = Util.getTld(hostname);

// tld is .com or .co.uk which means we means that length is 1 to big
// .com => 2 & .co.uk => 3
// ['', 'localhost']
const splitTld = tld.split('.');

// get TLD + root domain
// "localhost"
return hostname.split('.').slice(-splitTld.length).join('.');
```

The reason why it works is .slice(-2) or whatever actually just returns array[0] if only 1 element is available.

Reply:
:o might be the port, let me check that out. Good eye, I only checked real websites.

Reply:
@patrickhulce this is fixed now